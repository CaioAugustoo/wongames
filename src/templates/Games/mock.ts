import { QUERY_GAMES } from 'graphql/queries/games'

export const gamesMock = {
  request: {
    query: QUERY_GAMES,
    variables: { limit: 15 }
  },
  result: {
    data: {
      games: [
        {
          name: 'Strangeland - Official Soundtrack',
          slug: 'strangeland-official-soundtrack',
          cover: {
            url: 'https://res.cloudinary.com/ddnxfzdze/image/upload/v1623699186/strangeland_official_soundtrack_5c3f024f82.jpg'
          },
          developers: [{ name: 'Wormwood Studios', __typename: 'Developer' }],
          price: 6.49,
          __typename: 'Game'
        }
      ]
    }
  }
}

export const fetchMoreMock = {
  request: {
    query: QUERY_GAMES,
    variables: { limit: 15, start: 1 }
  },
  result: {
    data: {
      games: [
        {
          name: 'Fetch More Game',
          slug: 'fetch-more',
          price: 518.39,
          developers: [{ name: 'sample developer' }],
          cover: {
            url: 'sample-game.jpg'
          },
          __typename: 'Game'
        }
      ]
    }
  }
}
