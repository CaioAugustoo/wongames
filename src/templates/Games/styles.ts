import { Container } from 'components/Container'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Main = styled(Container)`
  ${({ theme }) => css`
    display: grid;

    ${media.greaterThan('medium')`
      grid-template-columns: 26rem 1fr;
      gap: ${theme.grid.gutter};
    `}
  `}
`
export const ShowMore = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    text-align: center;
    padding: ${theme.spacings.medium};
    text-transform: uppercase;
    font-weight: bold;
    cursor: pointer;

    > svg {
      color: ${theme.colors.primary};
    }
  `}
`
