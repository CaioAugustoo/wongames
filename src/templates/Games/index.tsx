import { GameCardProps } from 'components/GameCard'
import * as S from './styles'

export type GamesTemplateProps = {
  games?: GameCardProps[]
}

const GamesTemplate = ({ games = [] }: GamesTemplateProps) => (
  <S.Wrapper>
    <h1>Games</h1>
  </S.Wrapper>
)

export default GamesTemplate
