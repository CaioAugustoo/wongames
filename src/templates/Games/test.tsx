import { render, screen } from '@testing-library/react'

import Games from '.'

describe('<Games />', () => {
  it('should render the heading', () => {
    const { container } = render(<Games />)

    expect(screen.getByRole('heading', { name: /Games/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
