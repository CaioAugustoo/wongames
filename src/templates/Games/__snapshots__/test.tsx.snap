// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Games /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Games
  </h1>
</main>
`;
